specification/cognitiveservices/OpenAI.Inference/models/completions/tools.tsp

```diff
@@ -81,7 +81,23 @@ model ChatCompletionsFunctionToolDefinition
   type: "function";
 
   @doc("The function definition details for the function tool.")
-  function: FunctionDefinition;
+  @typeChangedFrom(ServiceApiVersions.v2024_08_01_Preview, FunctionDefinition)
```

Comment:
Can we add the `strict` field directly to `FunctionDefinition`?

Reply:
Just to centralize information: `FunctionDefinition` was originally introduced for usage under `ChatCompletionOptions.functions` where we provided a `FunctionDefinition[]`. The `functions` field has been [deprecated](https://platform.openai.com/docs/api-reference/chat/create#chat-create-functions) by OpenAI since. 

`tools` is the currently supported way of providing client side Function definitions. To techincally we originally made the mistake of reusing this model, but by looking at the OpenAPI definition from OpenAI, it does seem that the models are decoupled and only the one used in `tools` is being used. I think this is a good chance to decouple the usage of the model, so we can maintain support for the deprecated approach for as long as the service supports it while extending `tools`.